pyscreeze/__init__.py

```diff
@@ -495,32 +517,36 @@ def _screenshot_linux(imageFilename=None, region=None):
     """
     TODO
     """
-    if not scrotExists:
-        raise NotImplementedError('"scrot" must be installed to use screenshot functions in Linux. Run: sudo apt-get install scrot')
     if imageFilename is None:
         tmpFilename = '.screenshot%s.png' % (datetime.datetime.now().strftime('%Y-%m%d_%H-%M-%S-%f'))
     else:
         tmpFilename = imageFilename
-    if scrotExists:
-        subprocess.call(['scrot', '-z', tmpFilename])
-        im = Image.open(tmpFilename)
-
-        if region is not None:
-            assert len(region) == 4, 'region argument must be a tuple of four ints'
-            region = [int(x) for x in region]
-            im = im.crop((region[0], region[1], region[2] + region[0], region[3] + region[1]))
-            os.unlink(tmpFilename) # delete image of entire screen to save cropped version
-            im.save(tmpFilename)
+
+    if isWaylandSession:
+        if gnomescreenshotExists:
+            # scrot generates a blank black screen for Wayland sessions, but we can use gnome-screenshot if it's available
+            subprocess.call(['gnome-screenshot', '-f', tmpFilename])
         else:
-            # force loading before unlinking, Image.open() is lazy
-            im.load()
+            raise Exception("pyscreeze only supports Wayland session when gnome-screenshot is available")
```

Comment:
IMO it would be better to treat this failure here way above right when checking whether gnome-screenshot is available or not

Reply:
It seemed appropriate to minimize this patch by mimicking flow of logic used for `scrot`.  Handling the error condition inside a function also generates a more readable stack trace and affords us more flexibility when adding support for environments other than Gnome.

That said, I'm fine with throwing an exception earlier if @asweigart prefers that approach.